src/Lean/SubExpr.lean

```diff
@@ -0,0 +1,38 @@
+/-
+Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich, Daniel Selsam, Wojciech Nawrocki
+-/
+import Lean.Meta.Basic
+import Std.Data.RBMap
+
+namespace Lean
+
+/-- A position of a subexpression in an expression.
+
+See docstring of `SubExpr` for more detail.-/
+abbrev SubExpr.Pos := Nat
+
+/-- An expression and the position of a subexpression within this expression.
+
+Subexpressions are encoded as the parent root expression `e` and a
```

Comment:
In delaborator code, is it not the current expression rather than the root parent? E.g. in `Delaborator.SubExpr.descend` we set `expr := child`.

Reply:
Oooooh. Yes you are right. I assumed it worked like `Substring` where it stores the whole `String`.